sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -553,4 +578,233 @@ public DistroProperties resolveParentArtifact(Artifact parentArtifact, Server se
+	 * dependencies that start with specific prefixes such as {@code omod.}, {@code owa.}, {@code war},
+	 * {@code spa.frontendModule}, or {@code content.}. For each dependency, the method performs the following:</p>
+	 * <ul>
+	 *   <li>If the dependency is not present in {@code distroProperties}, it attempts to find the latest version
+	 *   matching the specified version range and adds it to {@code distroProperties}.</li>
+	 *   <li>If the dependency is present, it checks whether the version specified in {@code distroProperties}
+	 *   falls within the version range specified in {@code content.properties}. If it does not, an error is thrown.</li>
+	 * </ul>
+	 *
+	 * @param contentProperties The {@code Properties} object representing the {@code content.properties} file
+	 *                          of a content package.
+	 * @param distroProperties  The {@code DistroProperties} object containing key-value pairs specifying
+	 *                          the current distribution's dependencies and their versions.
+	 * @param zipFileName       The name of the ZIP file containing the {@code content.properties} file being processed.
+	 *                          Used in error messages to provide context.
+	 *
+	 * @throws MojoExecutionException If no matching version is found for a dependency not defined in
+	 *                                {@code distroProperties}, or if the version specified in {@code distroProperties}
+	 *                                does not match the version range in {@code content.properties}.
+	 */
+	protected void processContentProperties(Properties contentProperties, DistroProperties distroProperties, String zipFileName) throws MojoExecutionException {
+		for (String dependency : contentProperties.stringPropertyNames()) {
+			if (dependency.startsWith("omod.") || dependency.startsWith("owa.") || dependency.startsWith("war")
+					|| dependency.startsWith("spa.frontendModule") || dependency.startsWith("content.")) {
+				String versionRange = contentProperties.getProperty(dependency);
+				String distroVersion = distroProperties.get(dependency);
+
+				if (distroVersion == null) {
+					String latestVersion = findLatestMatchingVersion(dependency, versionRange);
+					if (latestVersion == null) {
+						throw new MojoExecutionException(
+								"No matching version found for dependency " + dependency + " in " + zipFileName);
+					}
+					distroProperties.add(dependency, latestVersion);
+				} else {
+					checkVersionInRange(dependency, versionRange, distroVersion, contentProperties.getProperty("name"));
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks if the groupId for a given dependency is overwritten in the distro properties.
+	 *
+	 * @param dependencyKey    The key of the dependency.
+	 * @param distroProperties The distro properties file.
+	 * @return The groupId to use for this dependency.
+	 */
+	private static String checkIfOverwritten(String dependencyKey, DistroProperties distroProperties) {
```

Comment:
Is there a reason we can't use the default `checkIfOverwritten()`?

Reply:
updated!